tests/Integration/Momento.Sdk.Tests/TopicTest.cs

```diff
@@ -0,0 +1,236 @@
+        
+        var testTask = Task.Run(async () =>
+        {
+            var messageCount = 0;
+            await foreach (var message in subscription)
+            {
+                Assert.NotNull(message);
+                Assert.True(message is TopicMessage.Item, $"Unexpected message: {message}");
+                if (((TopicMessage.Item)message).ValueByteArray!.SequenceEqual(skipMessage)) continue;
+                
+                Assert.Equal(valuesToSend[messageCount], ((TopicMessage.Item)message).ValueByteArray);
+
+                messageCount++;
+                if (messageCount == valuesToSend.Count)
+                {
+                    break;
+                }
+            }
+
+            subscription.Dispose();
+            return messageCount;
+        });
+        
+        // Send a few values to skip over to ensure the subscription received the messages under test
+        for (var i = 0; i < 5; i++)
+        {
+            await topicClient.PublishAsync(cacheName, topicName, skipMessage);
+            Thread.Sleep(100);
+        }
+        
+        foreach (var value in valuesToSend)
+        {
+            var publishResponse = await topicClient.PublishAsync(cacheName, topicName, value);
+            Assert.True(publishResponse is TopicPublishResponse.Success, $"Unexpected response: {publishResponse}");
+            Thread.Sleep(100);
+        }
+        foreach (var value in valuesToSend)
+        {
+            var publishResponse = await topicClient.PublishAsync(cacheName, topicName, value);
+            Assert.True(publishResponse is TopicPublishResponse.Success, $"Unexpected response: {publishResponse}");
+            Thread.Sleep(100);
+        }
+        
+        Assert.Equal(valuesToSend.Count, await testTask);
+    }
+
+    [Fact(Timeout = 5000)]
+    public async Task PublishAndSubscribe_String_Succeeds()
+    {
+        const string topicName = "topic_string";
+        var skipMessage = "skip";
+        var valuesToSend = new List<string>
+        {
+            "one",
+            "two",
+            "three",
+            "four",
+            "five"
+        };
+        
+        var subscribeResponse = await topicClient.SubscribeAsync(cacheName, topicName);
+        Assert.True(subscribeResponse is TopicSubscribeResponse.Subscription,
+            $"Unexpected response: {subscribeResponse}");
+        var subscription = (TopicSubscribeResponse.Subscription)subscribeResponse;
+        
+        var testTask = Task.Run(async () =>
+        {
+            var messageCount = 0;
+            await foreach (var message in subscription)
+            {
+                Assert.NotNull(message);
+                Assert.True(message is TopicMessage.Item, $"Unexpected message: {message}");
+                if (((TopicMessage.Item)message).ValueString!.Equals(skipMessage)) continue;
+                
+                Assert.Equal(valuesToSend[messageCount], ((TopicMessage.Item)message).ValueString);
```

Comment:
same here.

i don't mind casting in tests but I don't want it to be the UX.

If this pattern is just for tests then it might be worth adding an example `main` that shows what you are expecting it to look for users that are consuming it.

The number 1 thing that we have to get correct before we merge this PR is the user-facing API. Everything else can be tweaked/improved/cleaned up later - so you should make sure there is something in the PR that clearly illustrates what you are expecting that to look like for a consumer.